test/v2_integration.py

```diff
@@ -1584,39 +1583,44 @@ def parse_cert(order):
     return x509.load_pem_x509_certificate(order.fullchain_pem.encode(), default_backend())
 
 def test_admin_revoker_cert():
-    cert_file_pem = os.path.join(tempdir, "ar-cert.pem")
-    order = chisel2.auth_and_issue([random_domain()], cert_output=cert_file_pem)
+    cert_file = tempfile.NamedTemporaryFile(
+        dir=tempdir, suffix='.test_admin_revoker_cert.pem', delete=False)
+    cert_file.close()
+    order = chisel2.auth_and_issue([random_domain()], cert_output=cert_file.name)
     parsed_cert = parse_cert(order)
 
     # Revoke certificate by serial
     reset_akamai_purges()
-    run("./bin/admin-revoker serial-revoke --config %s/admin-revoker.json %x %d" % (
-        config_dir, parsed_cert.serial_number, 1))
+    run(["./bin/admin-revoker", "serial-revoke",
+        "--config", "%s/admin-revoker.json" % config_dir,
+        '%x' % parsed_cert.serial_number, '1'])
+
     # Wait for OCSP response to indicate revocation took place
-    ee_ocsp_url = "http://localhost:4002"
-    verify_ocsp(cert_file_pem, "/tmp/intermediate-cert-rsa-a.pem", ee_ocsp_url, "revoked")
+    verify_ocsp(cert_file.name, "/tmp/intermediate-cert-rsa-a.pem", "http://localhost:4002", "revoked")
     verify_akamai_purge()
 
 def test_admin_revoker_batched():
-    certs = []
-    serials = []
-    serialFile = os.path.join(tempdir, "serials.hex")
-    f = open(serialFile, "w")
-
-    for x in range(3):
-        cert_file_pem = os.path.join(tempdir, "ar-cert-%d.pem" % x)
-        certs.append(cert_file_pem)
-        order = chisel2.auth_and_issue([random_domain()], cert_output=cert_file_pem)
-        f.write("%x\n" % parse_cert(order).serial_number)
-    f.close()
+    serialFile = tempfile.NamedTemporaryFile(
+        dir=tempdir, suffix='.test_admin_revoker_batched.serials.hex',
+        mode='w', delete=False)
```

Comment:
Is there a reason this invocation has `mode='w'` and others don't?

Reply:
NamedTemporaryFile opens with `mode='w+b'` by default. Here we're writing text, not bytes, so it needs to be opened in not-binary mode. The others leave the mode out because they just close the file and let auth_and_issue reopen and write to it.